File: quantitative_test/celldie/celldie.rb
#!/usr/bin/env ruby
# 2018 Daniel Varga (vargad88@gmail.com)

def simulate_colony
    size = 1
    while size > 0 && size < 2000
        new_size = 0
        size.times { new_size += rand(0..3) }
        size = new_size
    end
    return size > 0
end

ITERATIONS=10000
dieout = 0
ITERATIONS.times {
    dieout += 1 if not simulate_colony
}

puts "chance of never die: #{100-dieout.to_f/ITERATIONS*100}"

File: quantitative_test/celldie/celldie.py
#!/usr/bin/env ruby
# 2018 Daniel Varga (vargad88@gmail.com)

import random

def simulate_colony():
    size = 1
    while size > 0 and size < 2000:
        new_size = 0
        for _ in range(size):
            new_size += random.randint(0,3)
        size = new_size
    return size > 0

iterations=10000
dieout = 0
for _ in range(iterations):
    if not simulate_colony(): dieout += 1

print("chance of never die: %f"%(100-float(dieout)/iterations*100))
